src/tools/wasm-fuzz-lattices.cpp

```diff
@@ -0,0 +1,389 @@
+      transferFunction.transfer(&(*cfgIter), yResult);
+      typename Lattice::Element zResult = z;
+      transferFunction.transfer(&(*cfgIter), zResult);
+
+      // Check monotonicity for every pair of transfer function outputs.
+      checkMonotonicity<Lattice>(
+        lattice, transferFuncName, &(*cfgIter), x, y, xResult, yResult);
+      checkMonotonicity<Lattice>(
+        lattice, transferFuncName, &(*cfgIter), x, z, xResult, zResult);
+      checkMonotonicity<Lattice>(
+        lattice, transferFuncName, &(*cfgIter), y, z, yResult, zResult);
+    }
+  }
+
+  // Checks properties of the LivenessTransferFunction on a randomly generated
+  // module function.
+  void checkLivenessTransferFunction(analysis::CFG& cfg, Function* func) {
+    analysis::FiniteIntPowersetLattice lattice(func->getNumLocals());
+    analysis::LivenessTransferFunction transferFunction;
+
+    check<analysis::FiniteIntPowersetLattice,
+          analysis::LivenessTransferFunction>(
+      cfg, lattice, transferFunction, "LivenessTransferFunction");
+  }
+
+  // Checks properties of the ReachingDefinitionsTransferFunction on a randomly
+  // generated module function.
+  void checkReachingDefinitionsTransferFunction(analysis::CFG& cfg,
+                                                Function* func) {
+    LocalGraph::GetSetses getSetses;
+    LocalGraph::Locations locations;
+    analysis::ReachingDefinitionsTransferFunction transferFunction(
+      func, getSetses, locations);
+    check<analysis::FinitePowersetLattice<LocalSet*>,
+          analysis::ReachingDefinitionsTransferFunction>(
+      cfg,
+      transferFunction.lattice,
+      transferFunction,
+      "ReachingDefinitionsTransferFunction");
+  }
+
+  // Generates a module. The module is used as an input to fuzz transfer
+  // functions as well as randomly generated lattice element states. Lattice
+  // properties are also fuzzed from the randomly generated states.
+  void run(uint64_t seed) {
+    // TODO: Reset the global type state to avoid monotonically increasing
+    // memory use.
+    RandEngine getRand(seed);
+    std::cout << "Running with seed " << seed << "\n";
+
+    // 4kb of random bytes should be enough for anyone!
+    std::vector<char> bytes(4096);
+    for (size_t i = 0; i < bytes.size(); i += sizeof(uint64_t)) {
+      *(uint64_t*)(bytes.data() + i) = getRand();
+    }
+
+    std::vector<char> bytes2 = bytes;
+
+    Module testModule;
+    TranslateToFuzzReader reader(testModule, std::move(bytes));
+    reader.build();
+    rand = Random(std::move(bytes2));
+
+    for (auto& func : testModule.functions) {
+      Function* currFunc = func.get();
+      // Imported functions can't be fuzzed here as they don't have a body.
+      if (currFunc->imported()) {
+        continue;
+      }
+      analysis::CFG cfg = analysis::CFG::fromFunction(currFunc);
+
+      checkLivenessTransferFunction(cfg, currFunc);
+      checkReachingDefinitionsTransferFunction(cfg, currFunc);
```

Comment:
To reduce work per seed, perhaps it would make sense to randomly pick one of these to do. On the other hand, maybe that's not worth it until we have a lot more lattices and analyses.